xbmc/pvr/guilib/guiinfo/PVRGUIInfo.cpp

```diff
@@ -478,7 +479,10 @@ bool CPVRGUIInfo::GetListItemAndPlayerLabel(const CFileItem* item, const CGUIInf
         break;
       }
       case LISTITEM_ICON:
-        if (!recording->Channel())
+        if (recording->ClientIconPath().empty() &&
+            recording->ClientThumbnailPath().empty() &&
+            !recording->Channel() &&
+            CServiceBroker::GetPVRManager().Providers()->GetNumProviders() > 1)
```

Comment:
1. Could you explain why > 1, not > 0? Maybe this is even worth a comment in the code?
2. Looking at the implementation of `CPVRRecording::Channel()`, this seems to be a quite expensive operation. I suggest to add the check as last term of the if clause.